drivers/serial/uart_mcux.c

```diff
@@ -73,6 +73,7 @@ FSL_FEATURE_UART_HAS_STOP_BIT_CONFIG_SUPPORT
 		return -ENOTSUP;
 	}
 
+#if defined(FSL_FEATURE_UART_HAS_MODEM_SUPPORT) && FSL_FEATURE_UART_HAS_MODEM_SUPPORT
```

Comment:
why is this guard being added?

Reply:
The uart_config_t structure members enableRxRTS and enableTxCTS are declared only if this condition is met. 
See [drivers/uart/fsl_uart.h](https://github.com/zephyrproject-rtos/hal_nxp/blob/862e001504bd6e0a4feade6a718e9f973116849c/mcux/mcux-sdk/drivers/uart/fsl_uart.h#L168).